src/commands.ts

```diff
@@ -343,6 +362,48 @@ export class CosmosDBCommands {
             }
         }
     }
+
+    public static async deleteDocDBDocument(doc: DocDBDocumentNode, explorer: CosmosDBExplorer): Promise<void> {
+        if (doc) {
+            const confirmed = await vscode.window.showWarningMessage("Are you sure you want to delete document '" + doc.label + "'?", "Yes", "No");
+            if (confirmed === "Yes") {
+                const masterKey = await doc.coll.db.getPrimaryMasterKey();
+                const endpoint = await doc.coll.db.getEndpoint();
+                const client = new DocumentClient(endpoint, { masterKey: masterKey });
+                const docLink = doc.getDocLink();
+                await new Promise((resolve, reject) => {
+                    client.deleteDocument(docLink, (err) => {
+                        err ? reject(new Error(err.body)) : resolve();
+                    });
+                });
+                explorer.refresh(doc.coll);
```

Comment:
Edge case - what if the user proposes to delete the document that is open? Should I check for that here and then close the result.json if that's the case?

Reply:
I see a '404, not found' error at a breakpoint ( `reject(err.body)` ) , but it doesn't bubble up. 
I'll discuss this offline.

Reply:
That's because you're not awaiting 'updateDocDBDocument' in extension.ts